types/global.d.ts

```diff
@@ -8,3 +11,9 @@ type RequestIdleCallbackDeadline = {
   readonly didTimeout: boolean;
   timeRemaining: () => number;
 };
+
+declare global {
+  interface HTMLElementEventMap {
+    [POINTER_DOWN_OUTSIDE]: PointerDownOutsideEvent;
+  }
+}
```

Comment:
It's necessary so that one of the overloads of `element.addEventlistener` is aware of the custom event types.

![image](https://user-images.githubusercontent.com/1539897/118810331-aad1fa80-b8a3-11eb-9616-60af2af2f0bc.png)

This works because of how they're typed:
![image](https://user-images.githubusercontent.com/1539897/118810418-c4734200-b8a3-11eb-907f-82348b17a41e.png)

See the `keyof HTMLElementEventMap`, that's

![image](https://user-images.githubusercontent.com/1539897/118810473-d81ea880-b8a3-11eb-8518-37e099dffaee.png)

and these are explicitly listed:

![image](https://user-images.githubusercontent.com/1539897/118810531-e40a6a80-b8a3-11eb-8692-4b5a75401926.png)

So effectively, we're just extending that Map with extra custom event types.

Reply:
We haven't had to do this for any of our other custom events though, e.g. in `FocusScope` we do this without issue:

```jsx
container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
```

Why is it an issue in your case?

Reply:
That's because they're not `CustomEvent`, they're just standard `Event`, as in they don't have anything special on`detail`.

Reply:
FYI, we've simplified things here by simply casting at the event listener call site.